src/ui/public/vis_maps/geohash_layer.js

```diff
@@ -0,0 +1,91 @@
+import KibanaMapLayer from './kibana_map_layer';
+import _ from 'lodash';
+import { EventEmitter } from 'events';
+import Heatmap from './heatmap';
+import ScaledCircles from './scaled_circles';
+import ShadedCircles from './shaded_circles';
+import GeohashGrid from './geohash_grid';
+
+export default class GeohashLayer extends KibanaMapLayer {
+
+  constructor(featureCollection, options, zoom, kibanaMap) {
+
+    super();
+
+    this._geohashGeoJson = featureCollection;
+    this._geohashOptions = options;
+    this._zoom = zoom;
+    this._kibanaMap = kibanaMap;
+
+    this._createGeohashMarkers();
+  }
+
+  _createGeohashMarkers() {
+    const markerOptions = {
+      valueFormatter: this._geohashOptions.valueFormatter,
+      tooltipFormatter: this._geohashOptions.tooltipFormatter
+    };
+    switch (this._geohashOptions.mapType) {
+      case 'Scaled Circle Markers':
+        this._geohashMarkers = new ScaledCircles(this._geohashGeoJson, markerOptions, this._zoom, this._kibanaMap);
+        break;
+      case 'Shaded Circle Markers':
+        this._geohashMarkers = new ShadedCircles(this._geohashGeoJson, markerOptions, this._zoom, this._kibanaMap);
+        break;
+      case 'Shaded Geohash Grid':
+        this._geohashMarkers = new GeohashGrid(this._geohashGeoJson, markerOptions, this._zoom, this._kibanaMap);
+        break;
+      case 'Heatmap':
+        this._geohashMarkers = new Heatmap(this._geohashGeoJson, {
+          radius: +this._geohashOptions.heatmap.heatRadius,
```

Comment:
it coerces it to the number type. It was like that before. Agreed it looks weird and is probably not necessary, I will change it.